h/views/admin_groups.py

```diff
@@ -201,6 +203,10 @@ def _template_context(self):
         }
 
 
+def _userid(username, authority):
```

Comment:
I don't _love_ this—all ears if there's a better way to glom together a `userid` without having to repeat the `models.User...` stuff verbatim each time.

Reply:
I don't think this writes anything to the DB so it's just instantiating a simple Python class - no significant cost.

If you don't want to have to instantiate a `User` object just to get a `userid` then we _could_ make a `format_userid` `@staticmethod` on `User`, and then `User.userid` would use that staticmethod, but other code could also call it, so it would be like `models.User.format_userid(username=username, authority=authority)`. I'm not sure that's much better than what we have now though.

The right solution may be to ask whether the code is needing for format userids without user objects in the first place...

Reply:
> I don't think this writes anything to the DB so it's just instantiating a simple Python class - no significant cost.

And that's why I did this instead of using the `user` service, which would have looked cleaner but been a DB hit.

I see this pattern pop up from time to time (the instantiate-a-user-model-to-compute-a-userid pattern) so I think it's worthwhile to create a helper at some point, yep. _That_ was what I wasn't thrilled about—that I had a little private method cluttering up this view, not the actual details of what it was doing :).

Reply:
> The right solution may be to ask whether the code is needing for format userids without user objects in the first place...

The service methods for modifying group members take userids, which is the pattern that other service methods take (IDs, not objects). The view's form collects usernames because they are entered by humans. We need a way to translate those usernames to userids. Yes, the view could retrieve corresponding user models out of the DB but there are two drawbacks to that:

* The service is going to be looking these models up subsequently, so it doubles the DB access
* It adds more DB logic to this already-getting-bloated view